core/keyboardshortcut_registry.js

```diff
@@ -0,0 +1,351 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview The namespace used to keep track of keyboard shortcuts and the
+ * key codes used to execute those shortcuts.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ */
+'use strict';
+
+goog.provide('Blockly.KeyboardShortcutRegistry');
+
+goog.require('Blockly.KeyboardShortcutItems');
+goog.require('Blockly.navigation');
+goog.require('Blockly.utils.object');
+
+
+/**
+ * Class for the registry of keyboard shortcuts. This is intended to be a
+ * singleton. You should not create a new instance, and only access this class
+ * from Blockly.KeyboardShortcutRegistry.registry.
+ * @constructor
+ */
+Blockly.KeyboardShortcutRegistry = function() {
```

Comment:
Do you forsee a different type of Shortcut registry. Curious if we could just call it the ShortcutRegistry?

Reply:
Not that I can really think of. I stuck with KeyboardShortcutRegistry because it was more descriptive but it is pretty cumbersome so happy to move to Blockly.ShortcutRegistry.

Reply:
done